### ISSUE TITLE
Add product size

### ISSUE THREAD

**Issue by mah11** [pombase/website#234]
The gene pages all have "...aa, ...kDa" for product size.

- Populate with aa counts and kDa for protein-coding genes.

- For non-coding RNA genes, just show the size in nucleotides (i.e. something like "product size: 247 nt"); they don't have amino acids and nobody really talks about their sizes in kDa.

This is one of the few things I still look upon the legacy site, so unless it's difficult I'd like to see it done soon. (If it is gnarly, remove the "next" label.)

**Comment by ValWood** [comment]
we might need to supply Kim with a calculation tables for amino acid molecular weights. I think this is part of Ensembl so Mark didn't need to do it.
It might be available as a web service though...

length is probably easier ;)

**Comment by Antonialock** [comment]
expasy has a tool
http://web.expasy.org/compute_pi/

**Comment by ValWood** [comment]
It would need to be a webservice, not a webinterface

**Comment by kimrutherford** [comment]
We should be able to calculate the product weight using BioPerl or EMBOSS.

The bigger job will probably take longer to get the sequences and store them in Chado.  There's an issue for that:
  https://github.com/pombase/pombase-chado/issues/568

At the moment all of that is done by Ensembl.

**Comment by mah11** [comment]
I'm told that the relevant BioPerl thing is Bio::Tools::SeqStats. Google says in BioPython it's ProtParam, and there are probably Java & Ruby equivalents out there somewhere as well.

Surely a web service would be overkill.

**Comment by kimrutherford** [comment]
I've added code to the loader to store the exon, UTR, transcript and protein sequences in Chado.  It also uses BioPerl to calculate the weight and stores that as a property.

> I'm told that the relevant BioPerl thing is Bio::Tools::SeqStats.

Thanks, that did the trick.

I'll now change the code that creates the data for the website to make the sequence and weight available for display.

(See also #96)

**Comment by kimrutherford** [comment]
All the code is now in place to show the product size and weight on the gene pages - if tonight's load works.

**Comment by kimrutherford** [comment]
> All the code is now in place to show the product size and weight on the gene pages - if tonight's load works.

The load worked!

But.

I've only just noticed that the weight is in daltons not kilodaltons.  I'll fix that now.

Also I forgot to deal with /codon_start / phase so there one gene (SPAC977.01) has the wrong weight.  I'll fix that.

I also didn't notice that the amino acid sequence includes the stop codon so the amino acid count is off by one.  I'll fix that too.

As a double check, I'm also going to get the pombe protein fasta from Ensembl and make sure the sequences match.

**Comment by kimrutherford** [comment]
> I've only just noticed that the weight is in daltons not kilodaltons. I'll fix that now.

Fixed.

> I also didn't notice that the amino acid sequence includes the stop codon so the amino acid count is off by one.

Fixed.

So the display is correct as far as I know for every gene except for SPAC977.01.  I'm still planning to compare to the current Ensembl protein fasta to be sure.

**Comment by kimrutherford** [comment]
> I'm still planning to compare to the current Ensembl protein fasta to be sure.

There are 37 differences between the output the from the latest load and the proteins from the last release from Ensembl.  I think I've accounted for some for those differences due bugs.  But have there been many gene edits since the release?  There are 20-ish that look like edits to the CDS locations in the contigs.  Is that a likely number?

Here's an example which is now out of frame in the latest Chado: SPAC9G1.15c

And SPAC110.05 seems to be SPAC110.06 now?

The protein sequence for SPAC823.02 is not in Chado but that gene is in the contig files - I'll look into that.

I didn't use the right translation table for the mitochondrial genome either.  Whoops.  Another thing to fix.

**Comment by ValWood** [comment]
- [ ] SPAC977.01 hasn't got a start codon, will fix that now...

**Comment by ValWood** [comment]
Oh I see,  SPAC977.01 is at the edge of a gap., I forgot.... It has  codon_start=3. I also added /partial for good measure...

**Comment by ValWood** [comment]
- [ ]  SPAC9G1.15c, mzt1 had somehow moved  out of frame (extra nucleotide stuck on the front), now fixed

**Comment by ValWood** [comment]
This had a new systematic Id assigned (quite a while ago). This is because  its a very small orf, and even though in the same genomic location, it is in a totally different feame with no sequence shared with the previous sequence. For this reason, I thought it better to assign a new systematic ID.

 /synonym="SPAC110.05"
/controlled_curation="term=warning, new gene; db_xref=PMID:24929437; date=20140704"
/controlled_curation="term=species distribution, Schizosaccharomyces pombe specific; date=20140820"
/controlled_curation="term=warning, replaced existing SPAC110.05 which was in the wrong frame; db_xref=PMID:24929437; date=20160317"
/product="Schizosaccharomyces pombe specific protein"
/colour=8
/systematic_id="SPAC110.06"

So this one  is fine.

**Comment by ValWood** [comment]
Probably the rest are largely these:
http://pombase2.aska.gen.nz/reference/PMID%3A26615217

I changed them recently so that the frameshift represents the exact amino acid sequence.
This  is better for users as they should see the correct sequence if translated, and better for me as it is clearer where I need to add/delete  the base(s) when the assembly is updated.
I think  they were represented a bit oddly before because Ensembl doesn't like frameshifts represented by overlaps IIRC.....

**Comment by ValWood** [comment]
we can double check that all are in our list here if you provide the iDs?
http://pombase2.aska.gen.nz/status/gene-coordinate-changes

(I know there is probably a better way of doing this long term, but I keep thinking we are near the end of gene coordinate updates, probably now worth wasting time in coming up with a new system. The important thing is that we have a record)

**Comment by ValWood** [comment]
- [ ] The protein sequence for SPAC823.02 is not in Chado but that gene is in the contig files - I'll look into that.

The original SPAC823.02 was deleted a while ago. It is coincident with what is now known to be tom6 . It is one that I got in the wrong frame. When I made the original prediction I did it from a short conserved intron. Because it did not resemble any part of the translation of the original I gave it a new ID SPAC823.17.  SPAC823.02 was then made a synonym  of SPAC823.17.

You can see the old arrangement here:
http://www.pombase.org/spombe/result/SPAC823.17
(shared intron, different frame scenario)

basically SPAC823.02	Chromosome I, 2583804-2583973 	43aa, 5.12 kDa
which is not in the contig file should  not appear

**Comment by ValWood** [comment]
SPAC823.02 should probably be removed as a synonym of SPAC823.17, its probably more confusing that helpful.
I'll change it to an obsolete ID.... it might help with the locus history to record it.
v

**Comment by ValWood** [comment]
added this warning to Tom6 instead
/controlled_curation="term=warning, This gene replaced SPAC823.02 which was previously annotated at this loci in the incorrect frame; db_xref=PMID:24929437; date=20170501"

**Comment by ValWood** [comment]
>>I've only just noticed that the weight is in daltons not kilodaltons. I'll fix that now.
>Fixed.

I think the point is in the wrong place? 
See any1
40144.1kDa
should be 40.14 kDa.
(show to 2 DP)

v

**Comment by kimrutherford** [comment]
> I think the point is in the wrong place?

Sorry, yet again I made the change but forgot to re-release.  Should be fixed now with a reload.

**Comment by kimrutherford** [comment]
> Also I forgot to deal with /codon_start / phase so there one gene (SPAC977.01) has the wrong weight.

Fixed for the next load.

> I didn't use the right translation table for the mitochondrial genome either.

Also fixed.

> http://pombase2.aska.gen.nz/status/gene-coordinate-changes

I checked the other differences and they are all in your coordinate changes list, except for:

SPAC12B10.09 - now shorter, 5' end
SPAC1556.06 - 3' end full of stop codons in Ensembl sequence, fixed now
SPAC22F3.11c and SPAC29A4.23 - exon/intron boundary changes

In those four cases the sequence in Chado matches what Artemis gives so I'm happy now.

I think we can call this done.

**Comment by mah11** [comment]
re

> added this warning to Tom6 instead
> /controlled_curation="term=warning, This gene replaced SPAC823.02 which was previously annotated at this loci in the incorrect frame; db_xref=PMID:24929437; date=20170501"

Did you commit it? I don't see it in my copy, and I just did svn update. Also, it should say "at this locus" - "loci" is the plural of "locus".

**Comment by ValWood** [comment]
fixed and committed.
